### PR TITLE
flatter glossary structure, expand built-ins and add error handling

### PR DESCRIPTION
**Pull request by depombo**
Solves https://github.com/alantech/docs/issues/34

### CODE REVIEW

File: src/error_handling.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    print(oneFifth.getOr(0.0))
+    // getOr can be expressed with an | operator
+    print(oneFifth | 0.0)
+  }
+
+  // checks if the Result has an error
+  const oneZeroth = reciprocal(0.0)
+  if oneZeroth.isErr() {
+    // gets the Result's Error or default if it is a value
+    print(oneZeroth | 1.2345)
```
Comment (dfellis) [inline src/error_handling.md:35]: This comment and the code inside don't match.

Recommend changing the `print` line to:

```alan
print(oneZeroth.getErr(noerr()))
```

File: src/builtins/functions.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#### Built-in Functions
```
Comment (dfellis) [inline src/builtins/functions.md:1]: :100: on breaking apart the built-in functions based on their logical groupings. SO much more readable!

File: src/builtins/operators.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+<table>
+  <thead>
+    <tr>
+      <td>Operator</td><td>Infix/Prefix</td><td>Precedence</td><td>Function</td>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td><code>?</code></td><td>infix</td><td>0</td><td>cond</td>
```
Comment (dfellis) [inline src/builtins/operators.md:13]: Now that this table is on a different page, putting a link to the function and/or the function description here is probably wise.

File: src/error_handling.md (hunk 1 above)
Comment (depombo) [reply src/error_handling.md]: thanks!

File: src/about_alan.md [hunk 4]
```diff
@@ -1,6 +1,6 @@
 # About Alan
 
-Alan is a natively-parallel, statically-compiled, type-inferred language with a familiar syntax and many compile-time and run-time safety guarantees.
+Alan is a implicitly-parallel, statically-compiled, event-based, type-inferred language with a familiar syntax and many compile-time and run-time safety guarantees.
```
[the remark this answers is not in the document]
Comment (depombo) [reply src/about_alan.md]: oops, thanks!

**Review by dfellis — APPROVED** [review]

**Pull request merged**